src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -745,7 +745,7 @@ public MethodSymbol ReduceExtensionMethod(TypeSymbol receiverType, CSharpCompila
                 throw new ArgumentNullException(nameof(receiverType));
             }
 
-            if (!this.IsExtensionMethod || this.MethodKind == MethodKind.ReducedExtension)
+            if (!this.IsExtensionMethod || this.MethodKind == MethodKind.ReducedExtension || receiverType.SpecialType == SpecialType.System_Void)
```

Comment:
This doesn't look like a change in SemanticModel. Where are instance members getting dropped?

Reply:
That check is here
https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Binder/Binder_Lookup.cs,1984

`SemanticModel.LookUpSymbols` first calls (binder.AddMemberLookupSymbolsInfo` to get regular members and then calls `MethodSymbol.ReduceExtensionMethod` for reduced extension methods
https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Compilation/CSharpSemanticModel.cs,1633